### PR TITLE
Implement EXTERNAL.

### PR DESCRIPTION
**Pull request by 1uc**

### CODE REVIEW

File: src/codegen/codegen_compatibility_visitor.cpp [hunk 1]
```diff
@@ -50,6 +50,22 @@ std::string CodegenCompatibilityVisitor::return_error_if_solve_method_is_unhandl
...
+    const std::shared_ptr<ast::Ast>& ast_node) const {
+    if (simulator == "neuron") {
+        // When generating code for NEURON, NMODL supports EXTERNAL variables,
+        // same as nocmodl.
+        return "";
+    }
+
+    // When generating code for CoreNEURON EXTERNAL variables aren't permitted.
+    auto external = std::dynamic_pointer_cast<ast::External>(ast_node);
+    return fmt::format("Found EXTERN at [{}] while generating code for CoreNEURON.\n",
```
Comment (JCGoran) [inline src/codegen/codegen_compatibility_visitor.cpp:65]: Slight nitpick: should be `EXTERNAL` here.

**Review by JCGoran — APPROVED** [review]
LGTM, only minor comments.

File: src/codegen/codegen_compatibility_visitor.cpp (hunk 1 above)
Comment (1uc) [inline src/codegen/codegen_compatibility_visitor.cpp:65]:
```suggestion
    return fmt::format("Found EXTERNAL at [{}] while generating code for CoreNEURON.\n",
```

Comment (1uc) [reply src/codegen/codegen_compatibility_visitor.cpp]: fixed, thanks.

File: src/codegen/codegen_neuron_cpp_visitor.cpp [hunk 2]
```diff
@@ -945,10 +953,35 @@ void CodegenNeuronCppVisitor::print_mechanism_global_var_structure(bool print_in
...
+        printer->fmt_line("return {};", var_expr);
+        printer->pop_block();
+    }
+    if (!codegen_global_variables.empty()) {
+        printer->add_newline();
+    }
+
+    for (const auto& var: info.external_variables) {
+        auto var_name = get_name(var);
+        printer->fmt_line("extern double {}();", var_name);
```
Comment (1uc) [inline src/codegen/codegen_neuron_cpp_visitor.cpp:978]:
```suggestion
        printer->fmt_line("double {}();", var_name);
```

**Pull request merged**